elmslie-compose/src/main/java/vivid/money/elmslie/compose/ElmComponentFragment.kt

```diff
@@ -0,0 +1,28 @@
+package vivid.money.elmslie.compose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rxjava3.subscribeAsState
+import androidx.fragment.app.Fragment
+import vivid.money.elmslie.android.screen.ElmDelegate
+import vivid.money.elmslie.android.screen.ElmScreen
+import vivid.money.elmslie.android.storeholder.LifecycleAwareStoreHolder
+
+abstract class ElmComponentFragment<Event : Any, Effect : Any, State : Any> : Fragment(),
+    ElmDelegate<Event, Effect, State> {
+
+    @Suppress("LeakingThis", "UnusedPrivateMember")
+    private val elm = ElmScreen(this, lifecycle) { requireActivity() }
+
+    protected val store
+        get() = storeHolder.store
+
+    override val storeHolder = LifecycleAwareStoreHolder(lifecycle, ::createStore)
+
+    final override fun render(state: State) = Unit
+
+    @Composable
+    fun state() = store.states.subscribeAsState(initial = store.currentState)
+
+    @Composable
+    fun effect() = store.effects.map { EffectWithKey(it) }.subscribeAsState(initial = null)
```

Comment:
Maybe we can make State to inherence from class ComposableState with field UiEffects. [And allow storing Composable side-effects inside this new field](https://developer.android.com/jetpack/compose/side-effects). In this case inside reducer, we will have our usual handle effects field for system effects, and this UIEffect field of State for Composable Ui Effects?

Reply:
Is this the same discussion that happened in slack after this sample was released? 
If i remember correctly the main problem with it was duplicating Effects after process death. Effect will be saved in state and dispatched you ui the second time after store recreation since there's no mechanism to remove it from state after it was displayed

Reply:
Hmm... Yes, now I've recollected this discussion. Let me think about it a little bit, I want to dive into these compose side-effects and understand pros and cons.

btw I'll prepare a PR with a fix of a race condition during the double start of a store this week, let's include this fix into the next version

Reply:
Of course, i'll stay in touch

Reply:
@opengamer29 What do you think about merging this mr for now and leave this improvement for a later discussion? I agree that this could be a great feature, but i don't really have an idea of how it could be implemented. In case you will come with a solution i would be happy to consider it

Reply:
Ok, let's merge it. I'm trying to find time to investigate how we can use compose side-effects, but it's not so simple to find time :(.
So it will be totally OK to merge this when especially if there will be an open issue for this improvement.

Reply:
I've created an [issue](https://github.com/vivid-money/elmslie/issues/46) for researching this problem